src/ipad54/netherblocks/listener/EventListener.php

```diff
@@ -2,13 +2,22 @@
 
 namespace ipad54\netherblocks\listener;
 
+use ipad54\netherblocks\utils\CustomIds;
+use pocketmine\block\VanillaBlocks;
 use pocketmine\event\Listener;
-use pocketmine\event\player\PlayerCreationEvent;
-use ipad54\netherblocks\player\Player as MyPlayer;
+use pocketmine\event\player\PlayerInteractEvent;
+use pocketmine\world\sound\FireExtinguishSound;
 
 class EventListener implements Listener {
 
-    public function onPlayerCreation(PlayerCreationEvent $event){
-        $event->setPlayerClass(MyPlayer::class);
-    }
+	public function onInteract(PlayerInteractEvent $event) : void {
+		$face = $event->getFace();
+		$target = $event->getBlock();
+		$block = $target->getSide($face);
+		$world = $target->getPosition()->getWorld();
+		if($block->getId() === CustomIds::SOUL_FIRE_BLOCK){

```

Comment:
You also need to check that the action equals PlayerInteractEvent::LEFT_CLICK_BLOCK